File: fix-urls.py
import os
import re

def walk(directory):
    for root, _, files in os.walk(directory):
        for file in files:
            if file.endswith('.ts') or file.endswith('.tsx'):
                yield os.path.join(root, file)

modified = 0
for filepath in walk('src'):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    original = content
    
    # We want to wrap uses of VITE_API_URL to strip 'undefined' strings safely without breaking the app.
    pattern1 = r'\(import\.meta(?: as any)?\)\.env\?\.VITE_API_URL'
    replacement1 = r"String((import.meta as any).env?.VITE_API_URL || '').replace('undefined', '')"
    
    pattern2 = r'(?<!\.)import\.meta\.env\.VITE_API_URL'
    
    content = re.sub(pattern1, replacement1, content)
    content = re.sub(pattern2, replacement1, content)
    
    if content != original:
        with open(filepath, 'w', encoding='utf-8') as f:
            f.write(content)
        modified += 1
        print(f"Patched {filepath}")

print(f"Successfully patched {modified} files.")

File: fix-urls.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(function(file) {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) { 
            results = results.concat(walk(file));
        } else if (file.endsWith('.ts') || file.endsWith('.tsx')) {
            results.push(file);
        }
    });
    return results;
}

const files = walk(path.join(__dirname, 'src'));
let modified = 0;

for (const file of files) {
    let content = fs.readFileSync(file, 'utf8');
    let original = content;
    
    // We want to wrap uses of VITE_API_URL to strip 'undefined' strings safely without breaking the app.
    // Instead of complex AST, we can just prepend a small global sanitizer right before the URL assignment.
    // Actually, string replacement of `(import.meta as any).env?.VITE_API_URL` to `((import.meta as any).env?.VITE_API_URL || '').replace('undefined', '')` 
    // is safe and robust!
    
    content = content.replace(/\(import\.meta( as any)?\)\.env\?\.VITE_API_URL/g, "String((import.meta as any).env?.VITE_API_URL || '').replace('undefined', '')");
    content = content.replace(/import\.meta\.env\.VITE_API_URL/g, "String((import.meta as any).env?.VITE_API_URL || '').replace('undefined', '')");
    
    if (content !== original) {
        fs.writeFileSync(file, content, 'utf8');
        modified++;
        console.log(`Patched ${path.basename(file)}`);
    }
}

console.log(`Successfully patched ${modified} files.`);
